### YamlMime:PythonClass
uid: azure.mgmt.recoveryservices.aio.operations.UsagesOperations
name: UsagesOperations
fullName: azure.mgmt.recoveryservices.aio.operations.UsagesOperations
module: azure.mgmt.recoveryservices.aio.operations
inheritances:
- builtins.object
summary: "> [!WARNING]\n> DO NOT instantiate this class directly.\n>\n> \n>\n> Instead,\
  \ you should access the following operations through\n>\n> <xref:azure.mgmt.recoveryservices.aio.RecoveryServicesClient>'s\n\
  >\n> <xref:usages> attribute.\n>"
constructor:
  syntax: UsagesOperations(*args, **kwargs)
methods:
- uid: azure.mgmt.recoveryservices.aio.operations.UsagesOperations.list_by_vaults
  name: list_by_vaults
  summary: Fetches the usages of the vault.
  signature: 'list_by_vaults(resource_group_name: str, vault_name: str, **kwargs:
    Any) -> AsyncIterable[VaultUsage]'
  parameters:
  - name: resource_group_name
    description: 'The name of the resource group. The name is case insensitive.

      Required.'
    isRequired: true
    types:
    - <xref:str>
  - name: vault_name
    description: The name of the recovery services vault. Required.
    isRequired: true
    types:
    - <xref:str>
  return:
    description: An iterator like instance of either VaultUsage or the result of cls(response)
    types:
    - <xref:azure.core.async_paging.AsyncItemPaged>[<xref:azure.mgmt.recoveryservices.models.VaultUsage>]
  exceptions:
  - type: azure.core.exceptions.HttpResponseError
attributes:
- uid: azure.mgmt.recoveryservices.aio.operations.UsagesOperations.models
  name: models
  signature: models = <module 'azure.mgmt.recoveryservices.models' from 'C:\\hostedtoolcache\\windows\\Python\\3.11.9\\x64\\Lib\\site-packages\\azure\\mgmt\\recoveryservices\\models\\__init__.py'>
